### ISSUE TITLE
QThreadPool.tryTake is not available in PyQt5=5.7 (used in debian9)

### ISSUE THREAD

**Issue by cpascual** [silx-kit/silx#3244]
Using the [debian9 backported package](http://www.silx.org/pub/debian/dists/binary/python3-silx_0.13.1-1~bpo9+1_amd64.deb) we ran into the following problem:

```
https://github.com/silx-kit/silx/issuesTraceback (most recent call last):  
File "/usr/lib/python3/dist-packages/silx/gui/plot/tools/profile/rois.py", line 237, in __regionChanged    
    self.invalidateProfile()  
File "/usr/lib/python3/dist-packages/silx/gui/plot/tools/profile/core.py", line 102, in invalidateProfile    
    profileManager.requestUpdateProfile(self)  
File "/usr/lib/python3/dist-packages/silx/gui/plot/tools/profile/manager.py", line 818, in requestUpdateProfile    
   if threadPool.tryTake(runner):
AttributeError: 'QThreadPool' object has no attribute 'tryTake'
Aborted (core dumped)
```

According to the Qt docs, [`QThreadPool.tryTake` was introduced in Qt=5.9](https://doc.qt.io/qt-5/qthreadpool.html#tryTake), but in debian9 it is 5.7

AFAIKT (according to the requirements in setup.py), PyQt=5.7 is supported by silx, and therefore the usage of `tryTake` should be replaced or at least protected. 
Otherwise, the requirements should specify  PyQt5>=5.9 **and some patch would be needed to fix the debian9 package**

**Comment by vallsv** [comment]
Hi.

Surprisingly I was working yesterday on a worker which could replace that code.

I see a way to replace that call by something else, as a work around.

Let us know if you expect a bug fix for the 0.13?

**Comment by cpascual** [comment]
> Let us know if you expect a bug fix for the 0.13?

Of course it would be nice, but I do not think that would be critical for us. We can develop and test using conda until it is ready. What isimportant for us is that at some point we can use the profile feature in a debian9

**Comment by cpascual** [comment]
Oh, and just re-reading my previous posts in this issue, I realized that I sound quite rude... 
It was not my intention!
It was just that I was writing in a hurry while multitasking. 
So, let me thank now @vallsv for the prompt reply and also let you know that if there is something I can do (test or work on a PR, just say so)

**Comment by vallsv** [comment]
No worry. Sounds like a normal bug report to me :-)

**Comment by t20100** [comment]
closed by PR #3250
